packages/mobile-sdk-alpha/src/client.ts

```diff
@@ -128,10 +128,31 @@ export function createSelfClient({ config, adapters }: { config: Config; adapter
     return adapters.analytics.trackEvent(event, payload);
   }
 
+  /**
+   * Retrieves the private key via the auth adapter.
+   * With great power comes great responsibility
+   */
+  async function getPrivateKey(): Promise<string | null> {
+    if (!adapters.auth) throw notImplemented('auth');
```

Comment:
is this check necessary if we're enforcing the auth adapter in required adapters anyway?